src/structures/ThreadChannel.js

```diff
@@ -0,0 +1,394 @@
+   *   .catch(console.error);
+   */
+  setArchived(archived = true, reason) {
+    return this.edit({ archived }, reason);
+  }
+
+  /**
+   * Sets the duration before the channel is automatically archived.
+   * @param {ThreadAutoArchiveDuration} autoArchiveDuration How long before the thread is automatically archived
+   * @param {string} [reason] Reason for changing the archive time
+   * @returns {Promise<ThreadChannel>}
+   * @example
+   * // Set the thread auto archive time to 1 hour
+   * thread.setAutoArchiveDuration(60)
+   *   .then(newThread => {
+   *     console.log(`Thread will now archive after ${newThread.autoArchiveDuration}`);
+   *    });
+   *   .catch(console.error);
+   */
+  setAutoArchiveDuration(autoArchiveDuration, reason) {
+    return this.edit({ autoArchiveDuration }, reason);
+  }
+
+  /**
+   * Sets whether the thread can be archived by anyone or just mods.
+   * @param {boolean} [locked=true] Whether the thread is locked
+   * @param {string} [reason] Reason for archiving or unarchiving
+   * @returns {Promise<ThreadChannel>}
+   * @example
+   * // Set the thread to locked
+   * thread.setLocked(true)
+   *   .then(newThread => console.log(`Thread is now ${newThread.locked ? 'locked' : 'unlocked'}`))
+   *   .catch(console.error);
+   */
+  setLocked(locked = true, reason) {
+    return this.edit({ locked }, reason);
+  }
+
+  /**
+   * Sets a new name for the thread.
+   * @param {string} name The new name for the thread
+   * @param {string} [reason] Reason for changing the thread's name
+   * @returns {Promise<ThreadChannel>}
+   * @example
+   * // Set a new thread name
+   * thread.setName('not_general')
+   *   .then(newThread => console.log(`Thread's new name is ${newThread.name}`))
+   *   .catch(console.error);
+   */
+  setName(name, reason) {
+    return this.edit({ name }, reason);
+  }
+
+  /**
+   * Sets the rate limit per user for this thread.
+   * @param {number} rateLimitPerUser The new ratelimit in seconds
+   * @param {string} [reason] Reason for changing the thread's ratelimits
+   * @returns {Promise<ThreadChannel>}
+   */
+  setRateLimitPerUser(rateLimitPerUser, reason) {
+    return this.edit({ rateLimitPerUser }, reason);
+  }
+
+  /**
+   * Whether the thread is editable by the client user (name, archived, autoArchiveDuration)
+   * @type {boolean}
+   * @readonly
+   */
+  get editable() {
+    return this.ownerID === this.client.user.id || this.manageable;
+  }
+
+  /**
+   * Whether the thread is joinable by the client user
+   * @type {boolean}
+   * @readonly
+   */
+  get joinable() {
+    return (
+      !this.archived &&
+      this.permissionsFor(this.client.user)?.has(
+        this.type === 'private_thread' ? Permissions.FLAGS.MANAGE_THREADS : Permissions.FLAGS.VIEW_CHANNEL,
+        false,
+      )
+    );
+  }
+
+  /**
+   * Whether the thread is manageable by the client user, for deleting or editing rateLimitPerUser or locked.
```

Comment:
Is the `, for deleting...` needed, since we'd have to update it if/when new properties are added

Reply:
The names don't exactly describe what they do, to me these are both management type things so I listed what you can actually do with each of them, can remove if desired.